src/main/java/org/opentripplanner/apis/gtfs/mapping/RouteRequestMapper.java

```diff
@@ -283,132 +110,616 @@ private static Stream<String> parseOperation(Collection<Map<String, Collection<S
+      );
+      preferences.withRental(rental -> setBicycleRentalPreferences(rental, args.getGraphQLRental())
+      );
+      setBicycleOptimization(preferences, args.getGraphQLOptimization());
+    }
   }
 
-  private static void setParkingPreferences(
-    CallerWithEnvironment callWith,
-    VehicleParkingPreferences.Builder parking
+  private static void setBicycleWalkPreferences(
+    VehicleWalkingPreferences.Builder preferences,
+    GraphQLTypes.GraphQLBicycleWalkPreferencesInput args
   ) {
-    callWith.argument("parking.unpreferredCost", parking::withUnpreferredVehicleParkingTagCost);
+    if (args != null) {
+      var speed = args.getGraphQLSpeed();
+      if (speed != null) {
+        preferences.withSpeed(speed);
+      }
+      var mountTime = args.getGraphQLMountDismountTime();
+      if (mountTime != null) {
+        preferences.withMountDismountTime(
+          DurationUtils.requireNonNegativeShort(mountTime, "bicycle mount dismount time")
+        );
+      }
+      var cost = args.getGraphQLCost();
+      if (cost != null) {
+        var reluctance = cost.getGraphQLReluctance();
+        if (reluctance != null) {
+          preferences.withReluctance(reluctance);
+        }
+        var mountCost = cost.getGraphQLMountDismountCost();
+        if (mountCost != null) {
+          preferences.withMountDismountCost(mountCost.toSeconds());
+        }
+      }
+    }
+  }
 
-    callWith.argument(
-      "parking.filters",
-      (Collection<Map<String, Object>> filters) -> {
-        parking.withRequiredVehicleParkingTags(parseSelectFilters(filters));
-        parking.withBannedVehicleParkingTags(parseNotFilters(filters));
+  private static void setBicycleParkingPreferences(
+    VehicleParkingPreferences.Builder preferences,
+    GraphQLTypes.GraphQLBicycleParkingPreferencesInput args,
+    DataFetchingEnvironment environment
+  ) {
+    if (args != null) {
+      var unpreferredCost = args.getGraphQLUnpreferredCost();
+      if (unpreferredCost != null) {
+        preferences.withUnpreferredVehicleParkingTagCost(unpreferredCost.toSeconds());
       }
+      var filters = getParkingFilters(environment, "bicycle");
+      preferences.withRequiredVehicleParkingTags(parseSelectFilters(filters));
+      preferences.withBannedVehicleParkingTags(parseNotFilters(filters));
+      var preferred = getParkingPreferred(environment, "bicycle");
+      preferences.withPreferredVehicleParkingTags(parseSelectFilters(preferred));
+      preferences.withNotPreferredVehicleParkingTags(parseNotFilters(preferred));
+    }
+  }
+
+  /**
+   * This methods returns required/banned parking tags of the given type from argument structure:
+   * preferences.street.type.parking.filters. This methods circumvents a bug in graphql-codegen as
+   * getting a list of input objects is not possible through using the generated types in
+   * {@link GraphQLTypes}.
+   * <p>
+   * TODO this ugliness can be removed when the bug gets fixed
```

Comment:
Is there a link to a ticket?

Reply:
I haven't created a ticket yet at least yet. Haven't really looked into this issue recently, so no clue how easily it could be fixed.